### PR TITLE
Bugfix for #117 - commander rank/ratings data updated from journal

### PR DESCRIPTION
**Pull request by Tkael**
#117
Add commander rating/rankings info from the journal to the commander object. Also add two new events... 'Empire promotion' and 'Federation promotion'.

### CODE REVIEW

File: JournalMonitor/JournalMonitor.cs [hunk 1]
```diff
@@ -331,6 +331,21 @@ public static List<Event> ParseJournalEntry(string line)
                                     events.Add(new ExplorationPromotionEvent(timestamp, rating) { raw = line });
                                     handled = true;
                                 }
+                                else if (data.ContainsKey("Federation"))
+                                {
+                                    Superpower superpower = Superpower.FromName("Federation");
+                                    data.TryGetValue("Federation", out val);
+                                    FederationRating rating = FederationRating.FromRank((int)(long)val);
```
Comment (richardbuckle) [inline JournalMonitor/JournalMonitor.cs]: Why the double cast here and below?

File: SpeechResponder/eddi.json [hunk 2]
```diff
@@ -461,6 +461,14 @@
       "name": "Docking timed out",
       "description": "Triggered when your docking request times out"
     },
+    "Empire promotion": {
+      "enabled": true,
+      "priority": 3,
+      "responder": true,
+      "script": "\"{if event.rank = \\\"Knight\\\":\\r\\n    sir\\r\\n|elif event.rank = \\\"Lord\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Baron\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Viscount\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Count\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Earl\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Marquis\\\":\\r\\n    my lord\\r\\n|elif event.rank = \\\"Duke\\\":\\r\\n    your grace\\r\\n|elif event.rank = \\\"Prince\\\":\\r\\n    your royal highness\\r\\n|elif event.rank = \\\"King\\\":\\r\\n    your majesty\\r\\n|else:\\r\\n    commander\\r\\n},\\r\\nthe Empirial Navy has \\r\\n{OneOf(\\\"granted you a promotion\\\", \\\"promoted you\\\", \\\"advanced you\\\", \\\"named you\\\")}\\r\\nto the rank of\\r\\n{event.rank}\\r\\n{Occasionally(2, \\\"for your service to the Empire\\\")}.      \"default\": true,",
```
Comment (richardbuckle) [inline SpeechResponder/eddi.json]: I build this branch and found that the Empir promotion script has gotten garbled. Looks like it has been double-escaped for JSON. Look at it in the app and you'll see.

File: JournalMonitor/JournalMonitor.cs (hunk 1 above)
Comment (Tkael) [reply JournalMonitor/JournalMonitor.cs]: mirroring other code snippets from JournalMonitor.cs. I'm guessing that jgm wasn't sure whether he'd be pulling a number or a string?

File: SpeechResponder/eddi.json (hunk 2 above)
Comment (Tkael) [reply SpeechResponder/eddi.json]: Done. Not sure how that happened?

**Pull request merged**